alacritty_terminal/src/term/search.rs

```diff
@@ -227,20 +244,30 @@ impl<T> Term<T> {
                 };
 
                 // Since we get the state from the DFA, it doesn't need to be checked.
-                state = unsafe { dfa.next_state_unchecked(state, byte) };
-            }
+                state = unsafe { regex.next_state_unchecked(state, byte) };
 
-            // Handle regex state changes.
-            if dfa.is_match_or_dead_state(state) {
-                if dfa.is_dead_state(state) {
-                    break;
-                } else {
-                    regex_match = Some(point);
+                // Matches require one additional BYTE of lookahead, so we check the match state for
+                // the first byte of every new character to determine if the last character was a
+                // match.
+                if i == 0 && regex.is_match_state(state) {
+                    regex_match = Some(last_point);
```

Comment:
it's not really obvious how it works, I guess the `last_point` is the nearest on to this cell and match is justified by the start of the next cell or when we reach the end?

Reply:
New regex-automata does lookaround and requires ingesting one additional byte before deciding if something is a match or not.

So we need to process one more byte, then if it is a match it was the last point that was an actual match.

Reply:
Yeah, I just wanted to ensure how it works.